rust/src/api.rs

```diff
@@ -150,8 +151,12 @@ pub async fn run_ldk(stream: StreamSink<Event>) -> Result<()> {
     tracing::debug!("Starting ldk node");
     let background_processor = wallet::run_ldk().await?;
     stream.add(Event::Ready);
-    // keep connection with maker alive!
-    connection::keep_alive().await?;
+
+    // spawn a connection task keeping the connection with the maker alive.
+    let peer_manager = wallet::get_peer_manager()?;
+    let connection_handle = connection::spawn(peer_manager);
+
+    try_join!(connection_handle)?;
```

Comment:
well I noticed if I don't await the handle, it will not do anything. Might be a blind spot in my rust logic. How can this be done better?

I figured a `try_join!` is a nice way. I also tried join, but this required me to process the result. Even though `connection::spawn` does not return a result i looks like `join` and `try_join` wrap them somehow in a `Result`

Reply:
> well I noticed if I don't await the handle, it will not do anything. Might be a blind spot in my rust logic. How can this be done better?

Are you sure? From the [docs](https://docs.rs/tokio/latest/tokio/task/fn.spawn.html):

> Spawns a new asynchronous task, returning a [JoinHandle](https://docs.rs/tokio/latest/tokio/task/struct.JoinHandle.html) for it.
>
> You do not have to .await the returned JoinHandle to make the provided future start execution. It will start running in the background immediately when spawn is called.

I would probably call `spawn` explicitly here and ignore the return type, rather than wrapping the `spawn` with a function.

Reply:
I tried again without `try_join!`, but it did not work. 😕